### ISSUE TITLE
VSCode Flutter Dart Very Slow initialization and code completition on Windows

### ISSUE THREAD

**Issue by DavidOrakpo** [Dart-Code/Dart-Code#5251]
When loading up a flutter project on VSCode on windows, I have to wait about 3 - 4 minutes, before flutter is initialized/loaded up. I have the experimental UI guidelines set to on, and it takes 3 - 4 minutes for them to be visible, which is one of the indicators I use to gauge when flutter is done initializing.  I have also run the same projects on both Mac and Linux machines and initialization is almost instantaneous. After the lengthy initialization, I still have to wait a minute or so for intellisense/code completion to kick in. 

I've attached below some screenshots from dart analyze
![Screenshot 2024-09-06 091718](https://github.com/user-attachments/assets/270f3103-8cac-41f8-8b71-afd5a60f3f4b)
![Screenshot 2024-09-06 091734](https://github.com/user-attachments/assets/212e250b-4699-4dae-ae2c-0ceaaa7aabc6)
![Screenshot 2024-09-06 091743](https://github.com/user-attachments/assets/8f5ea8d1-52fb-40c6-b5c0-776aeb59c9f6)

**To Reproduce**
Initialization of Flutter projects on windows

**Expected behavior**
Fast initialization and code completion like on MacOs and Linux machiness

**Screenshots**
Attached above

 - Operating System and version: Windows 11 23H2 22631.4037
 - VS Code version:  win 1.92.1
 - Dart extension version: 3.94.0
 - Dart/Flutter SDK version: 3.4.3/3.22.2
 - Target device (if the issue relates to Flutter debugging):

**Comment by DavidOrakpo** [comment]
I've attached below, dart analysis screenshots from my Linux machine running the same project. We can see that the timings/code completion is significantly faster on Linux.

Diagnostic Information on my Linux machine:
Dart Code extension: 3.96.0
Flutter extension: 3.96.0 (activated)
App Host: desktop
Version: linux 1.91.1

Workspace type: Flutter (LSP)
Workspace name: gate-creator-app

Dart (3.4.3): /home/sentry/fvm/versions/3.22.2/bin/cache/dart-sdk
Flutter (3.22.2): /home/sentry/fvm/versions/3.22.2 (sdk gphone64 x86 64 (android-x64/android))


![Uploading Screenshot from 
![Screenshot from 2024-09-06 11-02-26](https://github.com/user-attachments/assets/c93e552f-c78d-4de9-ad7d-50a943a9ac7b)
![Screenshot from 2024-09-06 11-02-32](https://github.com/user-attachments/assets/b4c89866-30fe-4bc1-8d1b-60f0624f4f48)
2024-09-06 11-00-30.png…]()

**Comment by DanTup** [comment]
@DavidOrakpo does this only occur on this specific project, or can you reproduce it on another? (If you can repro on a public project, that would really help).

If you see 3-4 minute delays when opening the project before the UI guidelines appear, my feeling is that this issue is probably not specific to code completion, and it's probably the analysis taking time. Usually when you invoke code completion, you have just typed something which will invalidate the analysis results for that file and it will require re-analyzing.

Could you try temporarily enabling the [instrumentation log file](https://dartcode.org/docs/logging/#analyzer-instrumentation), then open your project and leave it until the UI guides appear, then close VS Code and take a copy of the log file?

Sharing the log file may be tricky if this is a private project (it will contain parts of your files), but perhaps you can open the file and see if what (if any) kinds of events are being logged during that 3-4 minutes that might help understand what the server is doing.

The timing pages you included above might also be useful, but after only the slow initialization (without other requests), to see if things like the `initialized` notification show up with timings.

Thanks!

**Comment by DanTup** [comment]
Btw, there's a similar issue at https://github.com/dart-lang/sdk/issues/56307 about slow completion and the suspicion is that this is caused by the file being modified being depended on by a _lot_ of other files (directly or indirectly), and all of those need to be reanalyzed as you're changing a file (because your changes may introduce new errors on them). This might be part of your problem (of slow completion), but I don't think explained 3-4 minutes waiting for UI guides.

**Comment by DavidOrakpo** [comment]
> @DavidOrakpo does this only occur on this specific project, or can you reproduce it on another? (If you can repro on a public project, that would really help).
> 
> If you see 3-4 minute delays when opening the project before the UI guidelines appear, my feeling is that this issue is probably not specific to code completion, and it's probably the analysis taking time. Usually when you invoke code completion, you have just typed something which will invalidate the analysis results for that file and it will require re-analyzing.
> 
> Could you try temporarily enabling the [instrumentation log file](https://dartcode.org/docs/logging/#analyzer-instrumentation), then open your project and leave it until the UI guides appear, then close VS Code and take a copy of the log file?
> 
> Sharing the log file may be tricky if this is a private project (it will contain parts of your files), but perhaps you can open the file and see if what (if any) kinds of events are being logged during that 3-4 minutes that might help understand what the server is doing.
> 
> The timing pages you included above might also be useful, but after only the slow initialization (without other requests), to see if things like the `initialized` notification show up with timings.
> 
> Thanks!

Good afternoon Dan! Thanks so much for the reply! Yes, I can provide a public repo for testing. I have also used this repo to perform the tests I'd share in google drive below.   
Repo: https://github.com/DavidOrakpo/czarChallenge
Google Drive with Tests: https://drive.google.com/drive/folders/1t8ZQW_TVW_mzKY_vVZsDOqD5DP81Rcw8?usp=sharing


I performed the tests you suggested on 4 devices; Two Windows Laptops, A macbook and a Laptop running LInux mint.

From the results, we can see that the Linux and mac timings are exponentially faster than both windows based laptops. I've also attached the log files for each test as required. 


As for code completition, this is also instantaenous on non windows systems for me. Yes some projects I work on are very large, but this has never affected my mac or linux machines, only windows.

**Comment by DanTup** [comment]
@DavidOrakpo thanks! Unfortunately I haven't been able to reproduce this on my Windows PC using your project and the same version of Flutter. Looking at one of your logs (Lenovo), I see the server was started at 16:22:09 GMT and the first `publishDiagnostics` event is at 16:22:46.187, so it looks like there's a little over a 30s delay (which suspiciously is what https://github.com/dart-lang/sdk/issues/55621 was, however it doesn't seem like you have any analyzer plugins enabled, and I'm fairly sure that issue was specific to plugins).

Some questions/things to try:

1. Does it happen if you move the SDK outside of the workspace (it looks like your SDK is inside the `.fvm` folder inside the workspace)?
2. You're using Flutter 3.16 - there have been three main releases since then - are you able to test with the 3.24.2 release to see whether this changes anything (there have been many fixes in the Dart analyzer across those releases and it's possible you're hitting an issue that's been fixed)
3. Have you used Dart/Flutter a lot on both of the Windows machines you tested with, or is one of them the first time for this testing? I ask because in the past we've seen some perf issues on Windows when the `C:\Users\USERNAME\AppData\Local\.dartServer\.analysis-driver` folder gets large. It should automatically be pruned, but sometimes closing your IDE and wiping out that folder will yield improvements (although note that the first analysis of each project may be slow, because it will have to rebuild cached information that was stored in there)

Thanks!

**Comment by DavidOrakpo** [comment]
Damn. I get the same slow start ups with or without fvm. As for the flutter version, I can confirm that the slowdown occurs on the latest version of flutter, the repo I shared just happened to be on 3.16.

I do make frequent use of windows the most as it's my daily driver. It's why I want to get to the bottom of what is causing the slow initializations

**Comment by DanTup** [comment]
Thanks for testing/confirming - are you able to try clearing out the `.analysis-driver` folder?

Could you also see how long `flutter analyze` takes run from the root of that folder too?

And can you confirm whether it's _only_ this project with this issue, or if other projects of a similar size show the same? If it's only this project, then it may be interesting to make a copy of the project and trying to trim it down to see what inside it is causing the issue (for example, is the `build` folder huge? does deleting that make any difference?).

**Comment by DavidOrakpo** [comment]
I ran flutter analyze and got no issues. I deleted the .analysis-driver folder and retried my main work project. Still very slow initializations. I attached the timings below

![Screenshot 2024-09-09 210343](https://github.com/user-attachments/assets/cfaf2f04-7a80-40f8-a0e9-8dfeb74cd806)
![Screenshot 2024-09-09 210337](https://github.com/user-attachments/assets/20cfd297-4a51-4ad9-a8f1-0b98e0cda488)
![Screenshot 2024-09-09 210014](https://github.com/user-attachments/assets/539fb8a8-57a3-44db-8bba-2b61d55b026a)

**Comment by DanTup** [comment]
Thanks for testing - there's another log file the server has that I'm not super-familiar with, but might be useful here. If you open your **Workspace Settings** and add:

```js
"dart.analyzerAdditionalArgs": [
	"--analysis-driver-log=file:C:\\Dev\\Test Projects\\flutter_counter\\logs\\analysis_driver.txt"
]
```

But change the path to something valid (it should be prefixed with `file:` but otherwise a standard file path, with backslashes escaped). Saving the settings file will automatically restart the analysis server and begin logging. The file will probably get very big very quickly, so avoid interacting with the editor much, just let it do the initial analysis (until ui guides appear) and then close the editor and take a copy of the file (then you can remove it from your settings).

This log appears to log every file being analyzed and the time it takes, which might help understand where the time is going.

**Comment by DavidOrakpo** [comment]
@DanTup I've added a new folder, "Docs And Timings After VSCode Restart" with the needed documents. From my run, after the restart, it took 4 seconds for the UI guidelines to show which is a significant improvement. I hope the logs can help you out

**Comment by DanTup** [comment]
@DavidOrakpo Are startups all just a few seconds now? What about if you remove the logging?

I was hoping for a log file from a run that was taking 30-40s to start up (to understand why it was taking so long), but all of your recent logs have been < 5s. Is the issue gone?

**Comment by DavidOrakpo** [comment]
> @DavidOrakpo Are startups all just a few seconds now? What about if you remove the logging?
> 
> I was hoping for a log file from a run that was taking 30-40s to start up (to understand why it was taking so long), but all of your recent logs have been < 5s. Is the issue gone?

Sadly not. The issue still persists. I added the logging to my settings, restarted my system, confirmed the logging was still there, and booted up again.
![Screenshot 2024-09-11 123338](https://github.com/user-attachments/assets/13402d2c-be45-45aa-bec8-c6903721c510)
![Screenshot 2024-09-11 123342](https://github.com/user-attachments/assets/b447bfde-e101-49f2-b58a-90fe51567ded)
[analysis_driver.txt.txt](https://github.com/user-attachments/files/16962960/analysis_driver.txt.txt)

**Comment by DanTup** [comment]
> Sadly not. The issue still persists. I added the logging to my settings, restarted my system, confirmed the logging was still there, and booted up again.

Is this happening only the first time after a reboot? Are you able to capture the same log file both the first and second time after a reboot?

(I think we've seen this before, where the first time after a reboot there was a huge difference in performance, but I'm currently struggling to find the issue/notes)

**Comment by DavidOrakpo** [comment]
From what I've seen from testing, it has very slow startup first after a reboot....If I close the vscode window and open the same project again, it loads up much faster. But this is only project specific. I can load up another flutter project and it'd take 3 plus minutes to start up. Below, area the timings of my main work project. Looking at about 3 mins of start up
![Screenshot 2024-09-11 123943](https://github.com/user-attachments/assets/5e5a9a26-caf3-4188-a0e7-8f638a59a24d)
![Screenshot 2024-09-11 123949](https://github.com/user-attachments/assets/e747a62c-3804-4ba6-b5e7-415ecb273666)

**Comment by DanTup** [comment]
Thanks - that's useful info. I tried your project after a reboot but it didn't occur. Are you able to capture the same log for both the first and second time after a reboot? I'm trying to understand if it's just slower doing the same work (for ex. the disk access is slow but caches make it faster the next time), or if it's doing more work the first time after a reboot.

Thanks!

**Comment by DavidOrakpo** [comment]
Just did the latest test. Here are timings before the reboot:
![Screenshot 2024-09-11 124840](https://github.com/user-attachments/assets/3bdc6f2a-f15a-431c-be4c-77ac16b690d5)
![Screenshot 2024-09-11 124845](https://github.com/user-attachments/assets/2f9541b6-e004-4386-9ad3-e4d35d386c56)

Here are the timings, the second time (Closing vscode and running again)
![Screenshot 2024-09-11 124929](https://github.com/user-attachments/assets/e5183c5d-64b0-473b-a382-ba6157d9196a)
![Screenshot 2024-09-11 124941](https://github.com/user-attachments/assets/a773306d-a3e3-41af-a839-73a1ab6dfdb7)

The log file:
[analysis_server_log.txt](https://github.com/user-attachments/files/16963153/analysis_server_log.txt)

Hope this helps!

**Comment by DanTup** [comment]
@DavidOrakpo it's the analysis-drive-log I was hoping for for both. I also found where this came up before, and apparently I reproduced it then (https://github.com/dart-lang/sdk/issues/52947#issuecomment-1877514843), so I'm gonna try again - if so, I may be able to compare these logs locally without you having to keep testing :)

**Comment by DanTup** [comment]
@DavidOrakpo unfortunately I'm failing to repro again now, but something from the other thread I noticed is that this seemed to be made worse by huge `build` and/or `.dart_tool` folders.

Could you try deleting both your `.dart_tool` and `build` folders, then re-running `pub get` to get back the packages, and then reboot and see if the times are faster on the first load?

**Comment by DavidOrakpo** [comment]
> @DavidOrakpo it's the analysis-driver-log I was hoping for for both. I also found where this came up before, and apparently I reproduced it then ([dart-lang/sdk#52947 (comment)](https://github.com/dart-lang/sdk/issues/52947#issuecomment-1877514843)), so I'm gonna try again - if so, I may be able to compare these logs locally without you having to keep testing :)

Ahh I see. Here's a new break down of this test with the required log file.

Timing After restart:
![Screenshot 2024-09-11 142827](https://github.com/user-attachments/assets/131dcc37-7227-4b31-9ac3-e45069792762)
![Screenshot 2024-09-11 142831](https://github.com/user-attachments/assets/4299cf7c-0a0d-4abe-8bc8-80f1cc8d6bd0)

Timing on Second Load:
![Screenshot 2024-09-11 143201](https://github.com/user-attachments/assets/54e20b90-d161-4858-b9f6-c1f215a671cf)
![Screenshot 2024-09-11 143207](https://github.com/user-attachments/assets/aba5de47-75a1-4472-8010-cf90ff17f898)

Analysis driver log:
[analysis_server_log.txt](https://github.com/user-attachments/files/16964343/analysis_server_log.txt)
[analysis_server_logg.txt](https://github.com/user-attachments/files/16964350/analysis_server_logg.txt)

**Comment by DavidOrakpo** [comment]
> @DavidOrakpo unfortunately I'm failing to repro again now, but something from the other thread I noticed is that this seemed to be made worse by huge `build` and/or `.dart_tool` folders.
> 
> Could you try deleting both your `.dart_tool` and `build` folders, then re-running `pub get` to get back the packages, and then reboot and see if the times are faster on the first load?

For this test, the timings and logs are below.
From quick observation, after deleting those folders, start up time was 35 seconds. That's faster than the 41 seconds before, but still not so great

![Screenshot 2024-09-11 144018](https://github.com/user-attachments/assets/cd74be99-9c7d-4a19-be61-c84fdad67cd4)
![Screenshot 2024-09-11 144022](https://github.com/user-attachments/assets/cda1c342-efe7-4fbc-a3b1-a99c141212ba)
[analysis_server_loggg.txt](https://github.com/user-attachments/files/16964442/analysis_server_loggg.txt)
[analysis_server_logg.txt](https://github.com/user-attachments/files/16964443/analysis_server_logg.txt)
[analysis_server_log.txt](https://github.com/user-attachments/files/16964444/analysis_server_log.txt)

**Comment by DanTup** [comment]
@DavidOrakpo thanks, but these are still the analyzer instrumentation log and not the driver log I was after :)

**Comment by DavidOrakpo** [comment]
My bad

[analysis_driver.txt.txt](https://github.com/user-attachments/files/16967024/analysis_driver.txt.txt)

**Comment by DanTup** [comment]
@DavidOrakpo can you confirm which log this is? I was hoping for that log for both the first and second times you opened the project after a reboot (so I can compare them to see if different work is being done each time, or if the same work is done but takes longer).

I _think_ this file is a first run, but it's _very_ different to the previous ones you shared, so capturing them both together on subsequent runs may be useful just to eliminate any other variables between them. Thank you!

**Comment by DavidOrakpo** [comment]
Here it is sorry 
[analysis_driver.txt.txt](https://github.com/user-attachments/files/16967206/analysis_driver.txt.txt)
Before
![Screenshot 2024-09-11 181151](https://github.com/user-attachments/assets/5a360079-ecd7-494b-8891-cb2cfa761c95)

After
![Screenshot 2024-09-11 181227](https://github.com/user-attachments/assets/ae1154b9-fe0b-4332-b9ba-b2661130e8ea)

**Comment by DanTup** [comment]
@DavidOrakpo I was hoping for two log files taken around the same time to eliminate all other variables, but the last two above are quite far apart. I was hoping you could:

- reboot
- open the project and wait for the slow initialization
- close VS Code
- take a copy of the analysis driver log file
- re-open VS Code and the project, and wait for the (fast) initialization
- take a copy of the analysis driver log file

So the result would be two analysis driver log files taken a few minutes apart, one fast, one slow. There's a lot of differences between the last two files you provided, but I don't know how much of them might be caused by other factors (such as working in between).

Thanks!

**Comment by DavidOrakpo** [comment]
So sorry about the many tests. I've done as instructed:
Slow Initialization logfile: 
[analysis_driver.txt.txt](https://github.com/user-attachments/files/16967346/analysis_driver.txt.txt)

Fast Initialization
[analysis_driver.txt.txt](https://github.com/user-attachments/files/16967351/analysis_driver.txt.txt)
 log file:

**Comment by DanTup** [comment]
@DavidOrakpo thanks! The slow log file was very small this time, but does show times of 34918ms. This suggests all the extra stuff in the logs last time was unrelated and it might indeed be a result of the file access being slower (and not that any additional work was done).

I will do some more digging (as well as try again to reproduce this). Thank you for your help!

**Comment by DanTup** [comment]
I've been trying all morning to try and repro this, but without any luck. So far the worst I've seen is approx 8s after a reboot for the `initialized` notification and around 2s on a warm reload:

![image](https://github.com/user-attachments/assets/c958d59e-1f12-4f21-a565-331f802dcfd5)

It's difficult to know whether I'm not hitting the same issue as you, or whether I'm seeing the same issue but on faster disks so it's less obvious.

Can you reproduce this issue on any other projects? Does it occur if you do a _clean clone_ of the project into a new folder (this may help figure out if there may be other things in the project on disk for your main project that I don't have, that might be influencing the results).

Can you also confirm what kinds of disks are in the Windows machines you've tested (I have fairly fast NVMe drives and if you don't, that might explain at least some of the difference in times I see).

**Comment by DavidOrakpo** [comment]
I'm using Samsung 980 pro ssds which are PCI gen 4 ssds. So that's most likely not the issue. I was also able to reproduce the issue when I re installed windows like I said earlier. I'm not sure as well what could be the cause. I'd look for another public repo I have to demo this

**Comment by DanTup** [comment]
> I'm using Samsung 980 pro ssds which are PCI gen 4 ssds

Indeed, this is likely not the difference 😄

![Screenshot 2024-09-16 144950](https://github.com/user-attachments/assets/96a2197e-7bf9-43a7-9ffd-27e98e22208e)

I wondered if fvm might be playing a part (particularly since its SDK is _inside_ the analysis root and isn't excluded with `dart.analysisExcludedFolders`), but I tried copying a Flutter SDK into there (I don' thave fvm set up), but still couldn't repro.

If you make a copy of your project and remove the `.fvm` folder completely and point the workspace at another SDK, do you still see the same? And/or if you exclude the `.fvm` folder with `dart.analysisExcludedFolders` does it change anything?

Will test with your new repo shortly - thanks!

**Comment by DavidOrakpo** [comment]
> > I'm using Samsung 980 pro ssds which are PCI gen 4 ssds
> 
> Indeed, this is likely not the difference 😄
> 
> ![Screenshot 2024-09-16 144950](https://private-user-images.githubusercontent.com/1078012/367782284-96a2197e-7bf9-43a7-9ffd-27e98e22208e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1MDQ5MzMsIm5iZiI6MTcyNjUwNDYzMywicGF0aCI6Ii8xMDc4MDEyLzM2Nzc4MjI4NC05NmEyMTk3ZS03YmY5LTQzYTctOWZmZC0yN2U5OGUyMjIwOGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MTZUMTYzNzEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzgzOGUwYzQ5Yzg0ODI4OTlkYjAwMGM1ZWZmMWMyOGRmZTUwYWNkZGYyODA1N2M4OTNlOWVlMzVkNGVmZTVmMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.stu6IVC4Fn9P5fKY-kTmfUy9IZ61SRggN82Trrzi7LA)
> 
> I wondered if fvm might be playing a part (particularly since its SDK is _inside_ the analysis root and isn't excluded with `dart.analysisExcludedFolders`), but I tried copying a Flutter SDK into there (I don' thave fvm set up), but still couldn't repro.
> 
> If you make a copy of your project and remove the `.fvm` folder completely and point the workspace at another SDK, do you still see the same? And/or if you exclude the `.fvm` folder with `dart.analysisExcludedFolders` does it change anything?
> 
> Will test with your new repo shortly - thanks!
> 
> **Edit:** Unfortunately no luck with the new repo - also only 7-8s here after a reboot and/or shutdown/clean start.
> 
> **Edit2:** I set up FVM to see if it made any difference (I wondered if maybe the SDK being "inside" the root, or being a symlink might be a factor), but again, no repro.

Just tried this. Made a copy, deleted the fvm folder, and added it as an exclusion. Still slow times.

![Screenshot 2024-09-16 174745](https://github.com/user-attachments/assets/d43f2bda-386a-4675-aef0-a77a6a2cbd7f)
![Screenshot 2024-09-16 174748](https://github.com/user-attachments/assets/1ca1c436-8ac8-48b2-9e60-71705f61bb2b)
[analysis_driver.txt.txt](https://github.com/user-attachments/files/17016654/analysis_driver.txt.txt)

On my legion that I formatted, I did a clean install. So no old files where present. Although, it may not have been as clean a test as I'd have liked. Microsoft did an auto download of some setting back ups, I'm not sure if that affected things or not.

**Comment by DavidOrakpo** [comment]
It's also important to point out, that I installed flutter and ran the previous repo on my sisters system, and it worked just fine! This was a few before I formatted mine. It's what gave me the encouragement to do a clean wipe

**Comment by DanTup** [comment]
> On my legion that I formatted, I did a clean install. So no old files where present. 

But what about in the clone of the Git repository? Did you reuse the same folder from disk that might have untracked files (like the `.dart_tool` folder and `build` folder), or did you do a clean clone? If it was reused, could you try a clean clone to see if that might be the difference?

>  ran the previous repo on my sisters system, and it worked just fine!

Does she want to swap? 😄

I don't suppose you can think of anything different between them that could be a factor?

**Comment by DavidOrakpo** [comment]
> > On my legion that I formatted, I did a clean install. So no old files where present.
> 
> But what about in the clone of the Git repository? Did you reuse the same folder from disk that might have untracked files (like the `.dart_tool` folder and `build` folder), or did you do a clean clone? If it was reused, could you try a clean clone to see if that might be the difference?
> 
> > ran the previous repo on my sisters system, and it worked just fine!
> 
> Does she want to swap? 😄
> 
> I don't suppose you can think of anything different between them that could be a factor?

I'd test this out and get back to you. As for the difference in systems, nothing comes to mind. Mine is a gaming system. Could that be an issue? I figured that'd make performance even better

**Comment by DavidOrakpo** [comment]
> > On my legion that I formatted, I did a clean install. So no old files where present.
> 
> But what about in the clone of the Git repository? Did you reuse the same folder from disk that might have untracked files (like the `.dart_tool` folder and `build` folder), or did you do a clean clone? If it was reused, could you try a clean clone to see if that might be the difference?
> 
> > ran the previous repo on my sisters system, and it worked just fine!
> 
> Does she want to swap? 😄
> 
> I don't suppose you can think of anything different between them that could be a factor?

For clarification, I should ensure that the project repo .dart_tool and build folders are deleted before testing yeah?

**Comment by DanTup** [comment]
> Mine is a gaming system. Could that be an issue? I figured that'd make performance even better

I wouldn't expect that to be a problem itself (as you say, if anything, machines designed for gaming tend to perform well). One thing this just made me think of though, is that I have the folder I was testing in in Windows Defender exclusions, so it's possible that has some performance impact. I will try changing this and do some more testing with your repros from above.

> For clarification, I should ensure that the project repo .dart_tool and build folders are deleted before testing yeah?

Yep, though you'll need to re-open VS Code and accept the prompt to re-fetch packages (or run `flutter pub get`) after deleting `.dart_tool` to ensure all the packages are still available, and to ensure all of the usual analysis summaries are cached (so that it's a fair test and this isn't being done on the first timing after the reboot).

**Comment by DavidOrakpo** [comment]
Good morning Dan! Some great news! In my testing, I tried to see if I could replicate the bug in Android Studio. Once I launched android studio, I got a prompt to add certain folders to the Microsoft Defender Exclusion list. I selected it. I then checked the list and added my flutter project folder to it and turned off real time protection.
I restarted my laptop, and launched my project. I got about 5 seconds of boot time! And this is on my heaviest project yet!

**Comment by DavidOrakpo** [comment]
I tried turning Real-time protection on and the issue persisted. Are there a suggested list of folders I can add to the exclusion?

**Comment by Reprevise** [comment]
Hello from the other long GitHub issue about this 👋🏻 
I think I can back this up, turning off real time protection significantly improves analyzer startup time to around 3-4s from the original 20-30s.

However this isn't a permanent solution as real time protection turns itself back on when restarting so a list of folders to exclude would be appreciated.

**Comment by DanTup** [comment]
Thanks both for the additional info. I did try putting the test projects in an unexcluded folder above but couldn't repro. However when you both posted this I realised that my PubCache was still excluded, so I tried again with that excluded any my time after a reboot increased from 8s to about 10s, so although slower, it's still nowhere near as bad as you see. If I can get to the point of it being 30s it would be much easier to try out changes to see if it can be made faster.

**I would not _recommend_** excluding any folders from defender because there are may be security implications (for example if you're installing third party packages from Pub of Git dependencies from untrusted entities, they could probably contain anything). You'll have to take responsibility for any changes you make in this area.

Myself, I have excluded (and I must stress I am very careful what I install, including reviewing code in externally provided repros I run 🙂):

- My main Dev folder (which includes my own projects and _some_ of the repros I download when trying to reproduce bugs - although by default I tend to put them in a temp folder that's not excluded)
- My PubCache (`C:\Users\danny\AppData\Local\Pub`)
- My `.dartServer` folder where the cached summaries live (`C:\Users\danny\AppData\Local\.dartServer`)
- My downloaded SDK folders (`C:\Dev\Tools\Dart`.. Flutter is already in my main Dev folder since it's a git clone)

As I type out this list, I realise the `.dartServer` folder was still excluded in my last test, so I will remove that exclusion and do some additional testing shortly.

**Comment by DanTup** [comment]
Looks like the `.dartServer` folder was the final thing impacting this. With all of those exclusions gone, I see a time much closer to you (just under 30s here).

![image](https://github.com/user-attachments/assets/82033634-b525-4737-b8a2-34c7d1e6be2e)

So I guess Defender is slowing down the first read of these files. I don't know how much we can do about this, but it gives me a repro I can use to try some things out. Thank you both for your help (and patience!) in narrowing this down.

**Comment by DavidOrakpo** [comment]
Thanks! I added the dart server file to the exclusion list and I got a start up time of 10 seconds. It's more than what it was with Real time protection off, but exponentially smaller than 3 minutes!

**Comment by DanTup** [comment]
Closing this because I think the issue is resolved and/or a dupe of https://github.com/dart-lang/sdk/issues/56755. I suspect the resolution will be better documentation about this, and an easier way to track down when this is the issue (in latest Flutter master, I've added a Bytestore Timings page to the analyzer diagnostics, which will show the timings for reading files that will help identify when slow file reading is the problem).

If anyone is still having bad performance that they don't think is caused by this, please open a new issue with details. Thanks!

**Comment by ldsouza93** [comment]
> Thanks both for the additional info. I did try putting the test projects in an unexcluded folder above but couldn't repro. However when you both posted this I realised that my PubCache was still excluded, so I tried again with that excluded any my time after a reboot increased from 8s to about 10s, so although slower, it's still nowhere near as bad as you see. If I can get to the point of it being 30s it would be much easier to try out changes to see if it can be made faster.
> 
> **I would not _recommend_** excluding any folders from defender because there are may be security implications (for example if you're installing third party packages from Pub of Git dependencies from untrusted entities, they could probably contain anything). You'll have to take responsibility for any changes you make in this area.
> 
> Myself, I have excluded (and I must stress I am very careful what I install, including reviewing code in externally provided repros I run 🙂):
> 
> * My main Dev folder (which includes my own projects and _some_ of the repros I download when trying to reproduce bugs - although by default I tend to put them in a temp folder that's not excluded)
> * My PubCache (`C:\Users\danny\AppData\Local\Pub`)
> * My `.dartServer` folder where the cached summaries live (`C:\Users\danny\AppData\Local\.dartServer`)
> * My downloaded SDK folders (`C:\Dev\Tools\Dart`.. Flutter is already in my main Dev folder since it's a git clone)
> 
> As I type out this list, I realise the `.dartServer` folder was still excluded in my last test, so I will remove that exclusion and do some additional testing shortly.

When I first opened projects, the initial analysis would only start after making a code change, and it was a bit slow. However, after that, everything worked smoothly. Once I added the exclusions that you suggested to Windows Defender, the analysis at the first load became almost instant. I'm very satisfied with this solution.

**Comment by moinkhan780** [comment]
Flutter initializing taking too long on windows 11.

**Comment by DanTup** [comment]
@moinkhan780 it's not clear what "Flutter initializing" means here. If you mean the notification that says "Initializing Flutter" inside VS Code, that's a different step to the analysis initialization being described here. Please open a new issue with more details about what you're seeing (and what kind of times). Thanks!

**Comment by rashidotm** [comment]
> Thanks both for the additional info. I did try putting the test projects in an unexcluded folder above but couldn't repro. However when you both posted this I realised that my PubCache was still excluded, so I tried again with that excluded any my time after a reboot increased from 8s to about 10s, so although slower, it's still nowhere near as bad as you see. If I can get to the point of it being 30s it would be much easier to try out changes to see if it can be made faster.
> 
> **I would not _recommend_** excluding any folders from defender because there are may be security implications (for example if you're installing third party packages from Pub of Git dependencies from untrusted entities, they could probably contain anything). You'll have to take responsibility for any changes you make in this area.
> 
> Myself, I have excluded (and I must stress I am very careful what I install, including reviewing code in externally provided repros I run 🙂):
> 
> * My main Dev folder (which includes my own projects and _some_ of the repros I download when trying to reproduce bugs - although by default I tend to put them in a temp folder that's not excluded)
> * My PubCache (`C:\Users\danny\AppData\Local\Pub`)
> * My `.dartServer` folder where the cached summaries live (`C:\Users\danny\AppData\Local\.dartServer`)
> * My downloaded SDK folders (`C:\Dev\Tools\Dart`.. Flutter is already in my main Dev folder since it's a git clone)
> 
> As I type out this list, I realise the `.dartServer` folder was still excluded in my last test, so I will remove that exclusion and do some additional testing shortly.

how about adding in defender a process exclusion for Code.exe 🙄

**Comment by DanTup** [comment]
> how about adding in defender a process exclusion for Code.exe 🙄

I'm more confident in the code in `.dartServer` being safe than _everything_ I open in VS Code. I think I'm very careful, but I do sometimes have to test code provided by "anonymous" users on GitHub. I put such projects outside of my excluded Dev folder.

**Comment by zikriirfandi** [comment]
I had the same experience, every time I open a project, I have to wait about 2-3 minutes in order for dart server started.

I am using Windows 11, VSCode, dart flutter updated to latest.

**Comment by ElsayedDev** [comment]
the same issue occurs with me

**Comment by kivocsa99** [comment]
also happens here on windows 11 , with vscode 
it takes about 3 min to load and recognize everything in the project (no initializing flutter in the notification of vs code )
and when anything on any line on any file and size it takes about 20 m to recognize everything

**Comment by kivocsa99** [comment]
it stays like this : 
![image](https://github.com/user-attachments/assets/8ff88f56-1019-4568-beeb-dd56fc548c35)

**Comment by MobiliteDev** [comment]
Hello,  
I’m not sure whether I should create a separate issue or simply comment here, but I’ve recorded my screen and captured logs while experiencing this issue. Additionally, I captured `fix-all` running slowly.  

Below, you’ll find the video, the log, and the output of `dart:collect diagnostic information`.

[logs.txt](https://github.com/user-attachments/files/18291121/logs2.txt)

https://github.com/user-attachments/assets/b4ac3597-2cd8-40c9-a501-b0a284afc55a

<details>

<summary><strong>Workspace Environment</strong></summary>

```text
Dart Code extension: 3.102.0
Flutter extension: 3.102.0 (activated)

App: Visual Studio Code
App Host: desktop
Version: win 1.96.2

Workspace type: Dart, Flutter (LSP)
Workspace name: pnv_pods_ui.group

Dart (3.5.0): ****versions\3.24.0\bin\cache\dart-sdk
Flutter (3.24.0): ****versions\3.24.0 (Windows (windows-x64/windows))

```

</details>

<details>

<summary><strong>Output from 'dart info'</strong></summary>

`****versions\3.24.0\bin\cache\dart-sdk\bin\dart.exe info`

If providing this information as part of reporting a bug, please review the information
below to ensure it only contains things you're comfortable posting publicly.

#### General info

- Dart 3.5.0 (stable) (Tue Jul 30 02:17:59 2024 -0700) on "windows_x64"
- on windows / "Windows 10 Pro" 10.0 (Build 22631)
- locale is fr-FR

#### Project info

- sdk constraint: '>=3.1.0 <4.0.0'
- dependencies: 
- dev_dependencies: melos
- elided dependencies: 1

#### Process info

| Memory | CPU | Elapsed time | Command line |
| -----: | --: | -----------: | ------------ |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |
|   0 MB |  -- |              | dart.exe     |

</details>

<details>

<summary><strong>Output from 'flutter doctor'</strong></summary>

`****versions\3.24.0\bin\flutter.bat doctor -v`

```text
[!] Flutter (Channel stable, 3.24.0, on Microsoft Windows [version 10.0.22631.4602], locale fr-FR)
    • Flutter version 3.24.0 on channel stable at ****versions\3.24.0
    ! Warning: `flutter` on your path resolves to D:\dart_sdks\flutter_versions\tools\exec\flutter, which is not inside your current Flutter SDK checkout at ****versions\3.24.0. Consider adding ****versions\3.24.0\bin to the front of your path.
    ! Warning: `dart` on your path resolves to D:\dart_sdks\flutter_versions\tools\exec\dart, which is not inside your current Flutter SDK checkout at ****versions\3.24.0. Consider adding ****versions\3.24.0\bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 80c2e84975 (5 months ago), 2024-07-30 23:06:49 +0700
    • Engine revision b8800d88be
    • Dart version 3.5.0
    • DevTools version 2.37.2
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at *****
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Enterprise 2022 17.7.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Enterprise
    • Visual Studio Enterprise 2022 version 17.7.34024.191
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2024.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11609105)

[√] VS Code (version 1.96.2)
    • VS Code at *****
    • Flutter extension version 3.102.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [version 10.0.22631.4602]

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by ElsayedDev** [comment]
> Sorry for the delays in responding. It would be better to file separate issues and to include as much info as possible. If you are able to share a repro project that you see poor performance on, that could help a lot.
> 
> One thing that can slow down analysis is when you have a large number of files that need to be reanalyzed when you change a given file. For example in the video above if `base.dart` is imported by thousands of other files, then some kinds of changes to `base.dart` could require reanalyzing all of those libraries.
> 
> Right now it's not particularly easy to see exactly which files were reanalyzed (and how long each took) in a situation like this, but it's something I'd like to improve to help narrow down these kinds of issues.

the same project work fine for other operating system. I tried It on Mac and it worked fine

**Comment by DavidOrakpo** [comment]
Did yall read the back and forth than Dan and I had before we discovered a fix? Read through the conversation and see that windows defender is the prime cause of the slow downs

> > Sorry for the delays in responding. It would be better to file separate issues and to include as much info as possible. If you are able to share a repro project that you see poor performance on, that could help a lot.
> > 
> > One thing that can slow down analysis is when you have a large number of files that need to be reanalyzed when you change a given file. For example in the video above if `base.dart` is imported by thousands of other files, then some kinds of changes to `base.dart` could require reanalyzing all of those libraries.
> > 
> > Right now it's not particularly easy to see exactly which files were reanalyzed (and how long each took) in a situation like this, but it's something I'd like to improve to help narrow down these kinds of issues.
> 
> the same project work fine for other operating system. I tried It on Mac and it worked fine

**Comment by MobiliteDev** [comment]
Yes, that's why I shared the Task Manager screenshot showing the Antimalware Service Executable's consumption. I wanted to provide additional data to this issue with a complete example.

Unfortunately, I can't make changes to Windows Defender settings on my own. However, I'll reach out to my company's infrastructure team to see if they can assist with this.

**Comment by andynewman10** [comment]
I am constantly experiencing this problem as well, and it definitely IS a Windows Defender issue.

Windows Defender (aka 'Antimalware Service Executable') will constantly slow down code analysis to a point where this is nearly 10x times slower than on my Mac, and I am not exagerating.

To verify that Windows Defender is interfering, you can launch the Task Manager, and you can actually see the crazy Windows Defender activity in terms of file analysis by using [Process Monitor](https://learn.microsoft.com/sysinternals/downloads/procmon). The process name of Windows Defender is `MsMpEng.exe`. On my system, the `C:\Users\USERNAME\AppData\Local\.dartServer\` folder size is about 1GB, but has 97,000 files, and I can see Windows Defender always sneaks in there. I commonly have to wait 5-10 minutes beforce I can start to work, and the problem happens each time VS code is open, and even if you do `Developer: Window reload`. I might try @DanTup 's suggestion of deleting the folder.

It is possible to configure Windows Defender folder exclusions. Adding a Workspace folder exclusion has so far proven completely ineffective, even though my workspace is rather large. I am currently testing a `C:\Users\USERNAME\AppData\Local\.dartServer\` folder exclusion. It seems already much better, but I need to test more.

**Comment by DavidOrakpo** [comment]
Initially when I did my own tests, exclusion folders brought the start up time down exponentially. It no longer does that for me. I found myself turning off windows defender real time protection whenever I wanted to work on windows. It became very annoying so I just use Linux now for my work. Smoothest flutter dev experience yet on Linux 

> I am constantly experiencing this problem as well, and it definitely IS a Windows Defender issue.
> 
> Windows Defender (aka 'Antimalware Service Executable') will constantly slow down code analysis to a point where this is nearly 10x times slower than on my Mac, and I am not exagerating.
> 
> To verify that Windows Defender is interfering, you can launch the Task Manager, and you can actually see the crazy Windows Defender activity in terms of file analysis by using [Process Monitor](https://learn.microsoft.com/sysinternals/downloads/procmon). The process name of Windows Defender is `MsMpEng.exe`. On my system, the `C:\Users\USERNAME\AppData\Local\.dartServer\` folder size is about 1GB, but has 97,000 files, and I can see Windows Defender always sneaks in there. I commonly have to wait 5-10 minutes before I can start to work, and the problem happens each time VS code is open, and even if you do `Developer: Window reload` (note that it does not just happen once in this morning ; if you open your workspace 10 times during the day, the problem will happen 10 times... it is this bad). I might try @DanTup 's suggestion of deleting the folder.
> 
> It is possible to configure Windows Defender folder exclusions. Adding a Workspace folder exclusion has so far proven completely ineffective, even though my workspace is rather large. I am currently testing a `C:\Users\USERNAME\AppData\Local\.dartServer\` folder exclusion. It seems already much better, but I need to test more.
>

**Comment by DanTup** [comment]
@andynewman10 

> note that it does not just happen once in this morning ; if you open your workspace 10 times during the day, the problem will happen 10 times...

That's odd, I only seemed to see the slow reads the first time the server read the files from disk after a reboot. Subsequent restarts of VS Code / the analysis server would be fast (I assumed Defender tracked what it had already scanned).


> Initially when I did my own tests, exclusion folders brought the start up time down exponentially. It no longer does that for me.

That's odd - does the CPU usage suggest Defender is still scanning when you see this?

**Comment by andynewman10** [comment]
> [@andynewman10](https://github.com/andynewman10)
> 
> > note that it does not just happen once in this morning ; if you open your workspace 10 times during the day, the problem will happen 10 times...
> 
> That's odd, I only seemed to see the slow reads the first time the server read the files from disk after a reboot. Subsequent restarts of VS Code / the analysis server would be fast (I assumed Defender tracked what it had already scanned).

The behavior of Windows Defender is so opaque. It seems to constantly scan for threats, in a totally paranoïd fashion.

> 
> > Initially when I did my own tests, exclusion folders brought the start up time down exponentially. It no longer does that for me.
> 
> That's odd - does the CPU usage suggest Defender is still scanning when you see this?

Yes, definitely. Both CPU and sometimes disk usage.

**Comment by DanTup** [comment]
> With folder exclusions, my main work flutter project has an initialize time of 34 seconds. Without, it's like 3 or 4 minutes. It's better, but still far from excellent. Without Defender/Real-Time-Protection, the initialization is basically instantanous, which is analogous to what I experience on macos and Linux

When you're testing this, are you timing it (with real time protection) multiple times? It's possible the first time after turning it back on, there is more scanning happening than on a subsequent run, so it may be useful to know if there's a difference on first vs non-first run with it enabled.

Can you also confirm which folder(s) you have excluded? There are three folders that I think could have an impact (to different extends):

- project folder
- `.analysis_driver` folder
- `PUB_CACHE` folder

If you haven't tried excluded all of them, it may be worth testing to see if that's any better (I presume exclusions are less of a worry than turning off real-time protection, and I think that might re-enable itself after a reboot anyway?).

**Comment by andynewman10** [comment]
Note it is possible to [limit Windows Defender CPU usage](https://techcommunity.microsoft.com/discussions/windows-security/limit-windows-defender-cpu-usage/1511280/replies/4270077), though not on Windows Home versions. Of course, that's a last-resort approach, and may not solve the high disk usage.

**Comment by DanTup** [comment]
> Note it is possible to [limit Windows Defender CPU usage](https://techcommunity.microsoft.com/discussions/windows-security/limit-windows-defender-cpu-usage/1511280/replies/4270077), though not on Windows Home versions.

If this just throttles the process, it might also result in longer analysis times, because it blocks the file reads while it scans (at least on drives that are not Dev Drives... on Dev Drives the scanning is done asynchronously, although I had some worse perf issues with Dev Drives where the entire disk would just stop responding for 30s randomly, so I migrated back away!)

**Comment by andynewman10** [comment]
There is a simple experience that can be made to show how badly the performance is affected by Windows Defender.

1) boot Windows 11
2) Install a web server, such as this one: [XAMPP](https://www.apachefriends.org/fr/index.html). The installation process is very simple.
3) Launch the XAMPP control panel, then start MySQL and Apache
4) Surf on http://localhost/phpmyadmin. It takes **10-12 seconds** for the page to be displayed. And yes, you read correctly.
5) Disable Windows Defender real-time protection or, much better, install Bitdefender (free edition). Installing another antivirus really completely disables Windows Defender, which is still a bit active if you just turn off realtime protection.
6) Reboot
7) Launch the XAMPP control panel, then start MySQL and Apache
8) Surf on http://localhost/phpmyadmin. It takes **half a second**.

**Comment by IsmailALLA2** [comment]
yes i deleted the **\AppData\Local\.dartServer\.analysis-driver** folder and now when i start my flutter project on vs-code it starting quickly, and i'm not having to wait few minutes to start wwriting code to get autocomplete .

**Comment by DanTup** [comment]
@robertonic @IsmailALLA2 if you restart VS Code, then reboot, and then open your project, does the slow performance come back or is it cured?

As noted a few comments up, Windows Defender can make reading these files very slow (usually just the first time they are read after a reboot), so it's possible that your fix is only temporary (although Windows Defender exclusions could improve things). If not, and a one-off delete seems to improve performance across reboots, then it's possible this folder wasn't being cleaned up/maintained as intended.

There's an issue about the Windows Defender performance issues at https://github.com/dart-lang/sdk/issues/56755

**Comment by robertonic** [comment]
Just tried after a reboot. Now I see that clearing the  analysis-driver folder before this doesn't seem to help much.  The message  "Running 'Dart' Formatter" appears for a very long time. Any changes aren't highlighted as errors initally, but I think once the dart formatter is ready, for a while, correct changes are highlighted as errors. After a while longer everything is sorted out.  Maybe clearing the folder reduces the time that correct changes are flagged as errors?

I'd guess that the long time starting the dart formatter is the real problem and that you are right. This is a Windows Defender issue.

**Comment by robertonic** [comment]
Just excluded the   analysis-driver folder from Windows Defender and rebooted. 

Still see the message "Running 'Dart' Formatter" for over a minute.

**Comment by robertonic** [comment]
It's a fairly large project and maybe it takes a minute or so for Dart formatter to do its analysis.

**Comment by robertonic** [comment]
Thanks. Just tried and get 

Timing statistics collected by the analysis driver scheduler since last reset.

(name: <scheduler>, count: 0, elapsed: 0:00:00.000000, elapsedSelf: -0:00:00.423616)
  (name: getUnitElement, count: 192, elapsed: 0:00:00.382987, elapsedSelf: 0:00:00.024560)
    (name: libraryContext, count: 192, elapsed: 0:00:00.358427, elapsedSelf: 0:00:00.270343)(bytesGet: 6358228, cycleCount: 707, libraryCount: 1096, libraryLoadCount: 1096)
      (name: libraryCycle, count: 192, elapsed: 0:00:00.088084, elapsedSelf: 0:00:00.087821)
        (name: fileState.refresh, count: 3, elapsed: 0:00:00.000263, elapsedSelf: 0:00:00.000257)
          (name: getUnlinkedUnit, count: 3, elapsed: 0:00:00.000006, elapsedSelf: 0:00:00.000006)
  (name: analyzeFile, count: 1, elapsed: 0:00:00.040629, elapsedSelf: 0:00:00.039669)
    (name: libraryContext, count: 1, elapsed: 0:00:00.000960, elapsedSelf: 0:00:00.000446)(bytesGet: 40215, cycleCount: 1, libraryCount: 37, libraryLoadCount: 37)
      (name: libraryCycle, count: 1, elapsed: 0:00:00.000514, elapsedSelf: 0:00:00.000514)

**Comment by DanTup** [comment]
@robertonic based on those numbers, it doesn't seem like analysis is taking long at all. Perhaps it's worth opening a new issue and including exact steps you're taking to see 1min format time (if possible, a screen recording might help). Thanks!
